src/net/connection/timeout_thread.rs

```diff
@@ -19,6 +20,7 @@ pub const TIMEOUT_POLL_INTERVAL: u64 = 1;
 /// 3. Check if the last time we have heard from them (received a packet from them) is greater than the amount of time considered to be a timeout.
 /// 4. If they have timed out, send a notification up the stack.
 pub struct TimeoutThread {
+    alive: Arc<AtomicBool>,
```

Comment:
Can you rename this to `shutdown_signal`?

Reply:
Done.